### PR TITLE
 [patch] [bug] CEECORE-686 Windows memfs fixed

### PR DESCRIPTION
**Pull request by scrajesh**
* Added custom (optional) path property to ServeIndex(). 
* It's mainly used to set Posix Path methods/props for MemoryFileSystem

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -91,453 +76,471 @@ function serveIndex(root, options) {
...
+  var normalize = pathLib.normalize
+  , extname = pathLib.extname
+  , sep = pathLib.sep
+  , join = pathLib.join
+  , resolve = pathLib.resolve;
+
+  // default Stylesheet
+  var defaultStylesheet = join(__dirname, 'public', 'style.css');
+  // default Template
+  var defaultTemplate = join(__dirname, 'public', 'directory.html');
```
Comment (jchip) [inline index.js]: these two are for the module to find its own static file from disk, therefore, they should use native path.join.

File: test/test.js [hunk 2]
```diff
@@ -485,15 +485,15 @@ describe('serveIndex(root)', function () {
 
   describe('when using custom handler', function () {
```
Comment (jchip) [inline test/test.js]: can we ensure existing tests pass as is please?  ie: make no changes to tests.

**Review by jchip — CHANGES_REQUESTED** [review]

File: index.js [hunk 3]
```diff
@@ -91,453 +76,471 @@ function serveIndex(root, options) {
...
-    // malicious path
-    if ((path + sep).substr(0, rootPath.length) !== rootPath) {
-      debug('malicious path "%s"', path);
-      return next(createError(403));
+  var responseHandler = opts.responseHandler || null;
+
+  /**
+   * Respond with text/html.
+   */
+  function html(req, res, files, next, dir, showUp, displayIcons, path, view, template, stylesheet, filesystem) {
```
Comment (jchip) [inline index.js]: nit: moving so much code around cause a huge diff.  If possible, please move the code to end so original code at line 105 remains where they were to help with reduce code changes.

File: index.js (hunk 1 above)
Comment (scrajesh) [reply index.js]: corrected and moved it back to global scope.

File: test/test.js (hunk 2 above)
Comment (scrajesh) [reply test/test.js]: removed the optional responseHandler and used same serveIndex.html/plain/json object notation for custom handlers.

File: index.js (hunk 3 above)
Comment (scrajesh) [reply index.js]: named the returning function and retained it as per the original order.
moved the return statement to the end of the function.
hope this reduces the difference considerably.

**Review by jchip — APPROVED** [review]

**Pull request merged**